lib/routes/token-codes.js

```diff
@@ -0,0 +1,66 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+'use strict'
+
+const errors = require('../error')
+const isA = require('joi')
+const validators = require('./validators')
+const HEX_STRING = validators.HEX_STRING
+
+module.exports = (log, db, customs) => {
+  return [
+    {
+      method: 'POST',
+      path: '/session/verify/token',
+      config: {
+        auth: {
+          strategy: 'sessionToken'
+        },
+        validate: {
+          payload: {
+            uid: isA.string().max(32).regex(HEX_STRING).required(),
+            code: isA.string().min(0).max(1024).required() // code length is configurable but lets add some sane min/max values
+          }
+        }
+      },
+      handler (request, reply) {
+        log.begin('session.verify.token', request)
+
+        const uid = request.payload.uid
+        const code = request.payload.code.toUpperCase()
+
+        customs.checkIpOnly(request, 'verifyTokenCode')
```

Comment:
Since we've authenticated with the sessionToken here, we have the user's email address available in `request.auth.credentials.email`, meaning it could use `customs.check` here rather than `customs.checkIpOnly`.  It's probably worth doing so, to avoid incorrectly rate-limited people who happen to be behind a shared IP.